### PR TITLE
Enable request option for editing the log hours

### LINKED ISSUE #804: Enable request option for editing the log hours
## Description
As a Volunteer,
I need to not have access to edit hours once they are logged, however I can request edit from admins within a week so that I can edit the hous.

### Update [Required]
- [x] Disable edit option for volunteer
- [x] Add request edit option
- [x] Disable the request edit option after a week
- [x] Send email notification to the administrator for edit request
- [x] Send email notification to the volunteer once his hours are edited by the administrator

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
4 hours

### PR DESCRIPTION
**Pull request by anjali-dhanuka**
# Description
Allow volunteers to request editing of log hours within a week.

Fixes #804 

# Type of Change:
- Code
- Quality Assurance
- User Interface
- Outreach
- Documentation

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

# How Has This Been Tested?
image
image
image
image

# Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by abhishekarya286** [comment]
Is it ready for review?

**Comment by abhishekarya286** [comment]
Fix the Travis errors. Its throwing NoSuchElementException for the tests you wrote. @anjali-dhanuka

File: vms/shift/views.py [hunk 1]
```diff
@@ -1,728 +1,751 @@
...
+                    })
+        except:
+            raise Http404
+
+def edit_request(request, shift_id, volunteer_id):
+    vol = get_volunteer_by_id(volunteer_id)
+    shift = get_shift_by_id(shift_id)
+    message = render_to_string('shift/request_edit.txt', {
+                               'volunteer_first_name': vol.first_name,
+                               'volunteer_last_name': vol.last_name,
```
Comment (abhishekarya286) [inline vms/shift/views.py]: Please add a link in the message too that will direct the admin to the approve/cancel request page.

File: vms/shift/tests/test_shiftHours.py [hunk 2]
```diff
@@ -30,96 +34,116 @@ def setUpClass(cls):
...
+        Test edit of the logged hours.
+        """
         self.register_dataset()
         completed_shifts_page = self.completed_shifts_page
         completed_shifts_page.go_to_completed_shifts()
 
         completed_shifts_page.edit_hours('10:00', '13:00')
-        self.assertEqual(completed_shifts_page.get_shift_start_time(),
-                         '10 a.m.')
+        self.assertEqual(completed_shifts_page.get_shift_start_time(), '10 a.m.')
```
Comment (abhishekarya286) [inline vms/shift/tests/test_shiftHours.py]: I guess this needs to be changed to support the new changes.

**Comment by anjali-dhanuka** [comment]
@abhishekarya286 It's not ready for review, I need to write and fix tests. But for that, I need proper email id, email structure so that I can write tests.

**Comment by abhishekarya286** [comment]
Should I close this one @anjali-dhanuka

**Pull request closed without merging**